src/pkg/cli/compose/config_detector.go

```diff
@@ -0,0 +1,52 @@
+package compose
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/DefangLabs/secret-detector/pkg/scanner"
+)
+
+// assume that the input is a key-value pair string
+func detectConfig(input string) (detectorTypes []string, err error) {
+	// Detectors check for certain formats in a string to determine if it contains a secret.
+	// 		aws_client_id: AWS Client ID
+	// 		github: GitHub Personal Access Token
+	// 		high_entropy_string: calculated high entropy (randomness) in a string
+	// 		keyword: key contains a keyword (e.g. KEY, PASSWORD, SECRET, TOKEN, etc.)
+	// 		url_password: passwords in URL format
+
+	// Some detectors allow additional configuration options.
+	// "3" is the entropy threshold value (0 = low entropy, 4 = high entropy).
+
+	// create a custom scanner config in json
+	jsonCfg := `{
+		"transformers": ["json"],
+		"detectors": ["aws_client_id", "github", "high_entropy_string", "keyword", "url_password"],
+		"detectors_configs": {"keyword": ["3"], "high_entropy_string": ["3"]}
+		}`
+	cfg, err := scanner.NewConfigFromJson(strings.NewReader(jsonCfg))
+	if err != nil {
+		return nil, errors.New("Failed to make a config detector: " + err.Error())
```

Comment:
Use `fmt.Errorf("…: %w", err)` to *wrap* the error to keep the inner error accessible.